### PR TITLE
Test artifacts not uploaded to the correct bucket

### PR DESCRIPTION
**Pull request by jlewi**
* We should be using the kubernetes-jenkins bucket not a bucket in kubeflow-ci
  because spyglass and testgrid are still using the kubernetes-jenkins bucket.

* kubeflow/testing#488 needs to be submitted first before this will work.

* Related to kubeflow/kfctl#58

* Related to kubeflow/testing#489 run_e2e_workflow.py should set junit_xml_file

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/kubeflow/4321)

### CODE REVIEW

**Comment by jlewi** [comment]
/test all

kubeflow/testing#488 has merged

**Comment by jlewi** [comment]
Failure is a test artifact.

```
util.py                     71 INFO     Error from server (InternalError): error when creating "/tmp/kout648068887": Internal error occurred: failed call
ing webhook "pilot.validation.istio.io": Post https://istio-galley.istio-system.svc:443/admitpilot?timeout=30s: No SSH tunnels currently open. Were the t
argets able to accept an ssh-key for user "gke-2cba5fc1aeba73621a79"?
util.py                     71 INFO     Error from server (InternalError): error when creating "/tmp/kout648068887": Internal error occurred: failed call
ing webhook "pilot.validation.istio.io": Post https://istio-galley.istio-system.svc:443/admitpilot?timeout=30s: No SSH tunnels currently open. Were the t
argets able to accept an ssh-key for user "gke-2cba5fc1aeba73621a79"?
```

All the junit files were uploaded; e.g.
https://gcsweb.k8s.io/gcs/kubernetes-jenkins/pr-logs/pull/kubeflow_kubeflow/4321/kubeflow-presubmit/1184662029063950336/artifacts/junit_kubeflow-presubmit-kfctl-go-iap-istio-4321-a4b5eee-0336-8cd4/

**Comment by jlewi** [comment]
TFJob smoke test failed
/test all

**Comment by jlewi** [comment]
Tests passed!
https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/kubeflow_kubeflow/4321/kubeflow-presubmit/1185300031180115968/

**Comment by jlewi** [comment]
Here's an image showing grouping in testgrid using the pattern "[\w-]+"
grouping

File: jeremy.yaml [hunk 1]
```diff
@@ -0,0 +1,1164 @@
+access: GRANTED
```
Comment (lluunn) [inline jeremy.yaml]: what's this file?

File: kubeflow/jupyter/tests/jupyter_test.py [hunk 2]
```diff
@@ -97,7 +97,9 @@ def send_request(*args, **kwargs):
   return r
 
 
-def test_jupyter(env, namespace):
+def test_jupyter(record_xml_attribute, env, namespace):
```
Comment (lluunn) [inline kubeflow/jupyter/tests/jupyter_test.py]: what is the type of record_xml_attribute?

File: jeremy.yaml (hunk 1 above)
Comment (jlewi) [reply jeremy.yaml]: Woops will remove.

File: kubeflow/jupyter/tests/jupyter_test.py (hunk 2 above)
Comment (jlewi) [reply kubeflow/jupyter/tests/jupyter_test.py]: Its a test fixture created by pytest.

**Comment by jlewi** [comment]
@lluunn addressed your comments.

**Pull request merged**